static/compiler.js

```diff
@@ -365,36 +525,51 @@ function Compiler(domRoot, origFilters, windowLocalPrefix, onChangeCallback, lan
         return state;
     }
 
-    function deserialiseState(state) {
+    function deserialiseState(state, compilers, defaultCompiler) {
+        console.log("[WINDOW] Deserialising state ...");
         if (state.hasOwnProperty('sourcez')) {
             cppEditor.setValue(LZString.decompressFromBase64(state.sourcez));
         } else {
             cppEditor.setValue(state.source);
         }
-        state.compiler = mapCompiler(state.compiler);
-        setCompilerById(state.compiler);
-        domRoot.find('.compiler_options').val(state.options);
-        // Somewhat hackily persist compiler into local storage else when the ajax response comes in
-        // with the list of compilers it can splat over the deserialized version.
-        // The whole serialize/hash/localStorage code is a mess! TODO(mg): fix
-        setSetting('compiler', state.compiler);
-        updateAsm(true);  // Force the update to reset colours after calling cppEditor.setValue
+
+        if (slots.length != 0) {
+            console.log("[WINDOW] Deserialisation : deleting existing slots ...");
+            while (slots.length != 0) {
+                delete_and_unplace_slot(slots[slots.length - 1]);
+            }
+        }
+
+        // Deserialise 
+        for (var i = 0; i < state.slotCount; i++) {
+            var newSlot = create_and_place_slot(compilers, defaultCompiler);
```

Comment:
Nitpick (I'm sorry!): Can you stick to the camelCase convention for functions and variables please?